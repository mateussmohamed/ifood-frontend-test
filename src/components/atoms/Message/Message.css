.message {
  font-weight: 300;
  font-size: 16px;
  margin: 10px 0;
}

.message.info {
  color: grey;
}

.message.error {
  color: red;
}
