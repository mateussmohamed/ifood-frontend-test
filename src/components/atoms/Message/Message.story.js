import { storiesOf } from '@storybook/react';
import React from 'react';
import Message from './index';

const stories = storiesOf('Message', module);

stories.add('whit text prop', () => (
  <div style={{ float: 'left' }}>
    <Message text="Dormindo e cantando :)" />
  </div>
)).add('whit text and type prop', () => (
  <div style={{ float: 'left' }}>
    <Message text="Unlaunched country" type="error" />
    <Message text="Select Filter to load featured playlists =)" type="info" />
  </div>
));
