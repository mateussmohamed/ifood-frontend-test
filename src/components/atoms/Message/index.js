import React from 'react';
import PropTypes from 'prop-types';
import './Message.css';

const Message = ({ text, type }) => (
  <div className={`message ${type}`}>{text}</div>
);

Message.defaultProps = {
  text: 'Select filter for show playlists.',
  type: 'info',
};

Message.propTypes = {
  text: PropTypes.string,
  type: PropTypes.string,
};

export default Message;
