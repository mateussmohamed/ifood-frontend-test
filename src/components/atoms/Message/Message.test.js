import React from 'react';
import renderer from 'react-test-renderer';
import Message from './index';

describe('Message', () => {
  test('should Message match to snapshot', () => {
    const component = renderer.create(<Message text="awesome text" />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('should when text prop render the correct text', () => {
    const component = renderer.create(<Message text="awesome text" />);
    expect(component.toTree().props.text).toBe('awesome text');
  });

  test('should when type prop render the correct type', () => {
    const component = renderer.create(<Message type="error" />);
    expect(component.toTree().props.type).toBe('error');
  });
});
