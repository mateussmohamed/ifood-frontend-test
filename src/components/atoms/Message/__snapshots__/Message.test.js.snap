// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message should Message match to snapshot 1`] = `
<div
  className="message info"
>
  awesome text
</div>
`;
